File: LeetCode Problems/901 to 1000/974. Subarray Sums Divisible by K/974. Subarray Sums Divisible by K.kt
class Solution {
    fun subarraysDivByK(nums: IntArray, k: Int): Int {
        var sum:Int = 0; var cnt:Int = 0; var rem:Int = 0;
        val prefix = IntArray(k);
        prefix[0]++;
        for (items in nums) {
            sum += items;
            rem = (sum % k + k)% k;
            cnt += prefix[rem];
            prefix[rem]++;
        }
        return cnt;
    }
}

File: LeetCode Problems/901 to 1000/974. Subarray Sums Divisible by K/974. Subarray Sums Divisible by K.cpp
class Solution {
public:
    int subarraysDivByK(vector<int> &nums, int k) {
        int sum = 0, cnt = 0, rem;
        vector<int> prefix(k, 0);
        prefix[0]++;
        for (int i = 0; i < nums.size(); i++) {
            sum += nums[i];
            rem = (sum % k + k) % k;
            cnt += prefix[rem];
            prefix[rem]++;
        }
        return cnt;
    }
};